homeassistant/components/netgear/config_flow.py

```diff
@@ -0,0 +1,154 @@
+    user_schema = {
+        vol.Optional(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+        vol.Optional(CONF_PORT, default=user_input.get(CONF_PORT, 0)): int,
+        vol.Optional(CONF_SSL, default=user_input.get(CONF_SSL, False)): bool,
+    }
+    user_schema.update(_ordered_shared_schema(user_input))
+
+    return vol.Schema(user_schema)
+
+
+def _ordered_shared_schema(schema_input):
+    return {
+        vol.Optional(CONF_USERNAME, default=schema_input.get(CONF_USERNAME, "")): str,
+        vol.Required(CONF_PASSWORD, default=schema_input.get(CONF_PASSWORD, "")): str,
+    }
+
+
+class NetgearFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the netgear config flow."""
+        self.placeholders = {
+            CONF_NAME: "",
+            CONF_HOST: DEFAULT_HOST,
+            CONF_PORT: DEFAULT_PORT,
+            CONF_USERNAME: DEFAULT_USER,
+        }
+
+    async def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+        if not user_input:
+            user_input = {}
+
+        if self.placeholders.get(CONF_URL):
+            user_input.update({CONF_URL: self.placeholders[CONF_URL]})
+            step_id = "link"
+            data_schema = _discovery_schema_with_defaults(user_input)
+        else:
+            step_id = "user"
+            data_schema = _user_schema_with_defaults(user_input)
+
+        return self.async_show_form(
+            step_id=step_id,
+            data_schema=data_schema,
+            errors=errors or {},
+            description_placeholders=self.placeholders or {},
+        )
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input and not self.placeholders.get(CONF_URL):
+            self.placeholders[CONF_URL] = user_input.get(CONF_URL)
+
+        if not user_input:
+            return await self._show_setup_form(user_input, errors)
+
+        url = None
+        if self.placeholders.get(CONF_URL):
+            url = self.placeholders.get(CONF_URL)
+        host = user_input.get(CONF_HOST)
+        port = user_input.get(CONF_PORT)
+        ssl = user_input.get(CONF_SSL)
+        username = user_input.get(CONF_USERNAME)
+        password = user_input[CONF_PASSWORD]
+
+        # Open connection and check authentication
+        try:
+            api = await self.hass.async_add_executor_job(
+                async_get_api, password, host, username, port, ssl, url
+            )
+        except CannotLoginException:
+            errors["base"] = "config"
+
+        if errors:
+            return await self._show_setup_form(user_input, errors)
+
+        # Check if already configured
+        infos = await self.hass.async_add_executor_job(api.get_info)
```

Comment:
@MartinHjelmare 
Since creating `async_get_api`, I get
```
    return await self.async_step_user(user_input)
  File "/Users/pollet/dev/home-assistant/core/homeassistant/components/netgear/config_flow.py", line 112, in async_step_user
    infos = await self.hass.async_add_executor_job(api.get_info)
AttributeError: 'coroutine' object has no attribute 'get_info'
```

here 🤔

Reply:
`async def async_get_api(` --> `def get_api(`